core/src/main/java/ai/timefold/solver/core/config/solver/EnvironmentMode.java

```diff
@@ -106,22 +130,46 @@ public enum EnvironmentMode {
         this.asserted = asserted;
     }
 
+    public boolean isStepAssertOrMore() {
+        return switch (this) {
+            case NON_REPRODUCIBLE, NO_ASSERT, PHASE_ASSERT, STEP_ASSERT, NON_INTRUSIVE_FULL_ASSERT, FULL_ASSERT,
+                    TRACKED_FULL_ASSERT ->
+                isAsserted() && this != PHASE_ASSERT;
+            case REPRODUCIBLE -> PHASE_ASSERT.isAsserted();
```

Comment:
Should it be `NO_ASSERT.isAsserted()` instead?

Reply:
I had to adjust quite a lot of test coverage for this.